plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBTransaction.java

```diff
@@ -85,14 +87,35 @@ public void remove(final SegmentIdentifier segmentId, final byte[] key) {
 
   @Override
   public void commit() throws StorageException {
+    commit(0, 3);
+  }
+
+  public void commitWithRetries(final int retryLimit) throws StorageException {
+    commit(0, retryLimit);
+  }
+
+  void commit(final int attemptNumber, final int retryLimit) throws StorageException {
     try (final OperationTimer.TimingContext ignored = metrics.getCommitLatency().startTimer()) {
       innerTx.commit();
     } catch (final RocksDBException e) {
-      if (e.getMessage().contains(NO_SPACE_LEFT_ON_DEVICE)) {
+      if (e.getMessage().contains(ERR_NO_SPACE_LEFT_ON_DEVICE)) {
         logger.error(e.getMessage());
         System.exit(0);
+      } else if (e.getMessage().equals(ERR_BUSY) && attemptNumber < retryLimit) {
+        logger.trace(
+            "RocksDB Busy exception caught on attempt {} of {}, retrying",
+            attemptNumber,
+            retryLimit);
+        commit(attemptNumber + 1, retryLimit);
```

Comment:
do you think a direct retry without any delay will fix the problem ? maybe setting the level to info like that we can debug and see if this fix can unblock this kind of situation ?

Reply:
I haven't yet been able to reproduce a busy exception, but I bumped the logging to warn.  It seems reasonable to leave it at that level since it likely will provide good signal and shouldn't spam the logs due to the transaction commit timeout.